src/main/scala/com/microsoft/ml/spark/lightgbm/TrainUtils.scala

```diff
@@ -60,6 +69,34 @@ private object TrainUtils extends Serializable {
     datasetPtr
   }
 
+  /**
+    * Sample the first several rows to determine whether to construct sparse or dense matrix in lightgbm native code.
+    * @param rowsIter  Iterator of rows.
+    * @param schema The schema.
+    * @param columnParams The column parameters.
+    * @return A reconstructed iterator with the same original rows and whether the matrix should be sparse or dense.
+    */
+  def sampleRowsForArrayType(rowsIter: Iterator[Row], schema: StructType,
+                             columnParams: ColumnParams): (Iterator[Row], Boolean) = {
+    var sampleRow = 0
```

Comment:
Do we need to look at all 10? perhaps just using the first value would give the same clue and allow a cleaner implementation

Reply:
discussed offline, cleaned up the code a bit more